File: app/src/main/java/com/selflearning/chemistree/chemistry/inorganic/bases/BasesDao.java
package com.selflearning.chemistree.chemistry.inorganic.bases;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

@Dao
public interface BasesDao {

    @Query("SELECT * FROM bases")
    List<Bases> getAllBases();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertBase(Bases base);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAllBase(List<Bases> bases);

}

File: app/src/main/java/com/selflearning/chemistree/chemistry/inorganic/bases/BasesDao.kt
package com.selflearning.chemistree.chemistry.inorganic.bases

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
interface BasesDao {

    @Query("SELECT * FROM bases")
    fun getAll(): LiveData<List<Bases>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertBase(base: Bases)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAllBase(bases: List<Bases>)
}